src/exp2.jl

```diff
@@ -0,0 +1,66 @@
+# Import the generated code
+for i=1:15
+    include("exp_generated/exp_$i.jl")
+end
+
+function getmem(cache,k)
+    return cache[k-1];
+end
+
+function _exp2!(A; caches=nothing, do_balancing = A isa StridedMatrix)
```

Comment:
I assume this is an internal function that shouldn't be called besides testing?

Reply:
I view `_exp2!` equivalent to `_exp!` or a replacement thereof. When we are done comparing `_exp!` and `_exp2!`, we could consider renaming `_exp!` <- `_exp2!`. The function `_exp!` is currently not exported.